odml/tools/rdf_converter.py

```diff
@@ -0,0 +1,141 @@
+                if terminology_node:
+                    self.g.add((curr_node, fmt.rdf_map(k), terminology_node))
+                else:
+                    # adding terminology to the hub and to link with the doc
+                    node = URIRef(odmlns + str(uuid.uuid4()))
+                    self.g.add((node, odmlns.Terminology, URIRef(terminology_url)))
+                    self.g.add((self.hub_root, odmlns.hasTerminology, node))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+            # generating nodes for entities: sections, properties and bags of values
+            elif (isinstance(fmt, format.Document.__class__) or
+                      isinstance(fmt, format.Section.__class__)) and \
+                            k == 'sections' and len(getattr(e, k)) > 0:
+                sections = getattr(e, k)
+                for s in sections:
+                    node = URIRef(odmlns + str(s.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(s, node)
+            elif isinstance(fmt, format.Section.__class__) and \
+                            k == 'properties' and len(getattr(e, k)) > 0:
+                properties = getattr(e, k)
+                for p in properties:
+                    node = URIRef(odmlns + str(p.id))
+                    self.g.add((curr_node, fmt.rdf_map(k), node))
+                    self.save_element(p, node)
+            elif isinstance(fmt, format.Property.__class__) and \
+                            k == 'value' and len(getattr(e, k)) > 0:
+                values = getattr(e, k)
+                bag = BNode(str(uuid.uuid4()))
+                self.g.add((curr_node, fmt.rdf_map(k), bag))
+                for v in values:
+                    self.g.add((bag, RDF.li, Literal(v)))
+            # adding entities' properties
+            else:
+                val = getattr(e, k)
+                if val is None or not val:
+                    continue
+                elif k == 'date':
+                    self.g.add((curr_node, fmt.rdf_map(k), Literal(val, datatype=XSD.date)))
+                self.g.add((curr_node, fmt.rdf_map(k), Literal(val)))
+        return self.g
+
+    def _get_terminology_by_value(self, url):
+        return self.g.value(predicate=odmlns.Terminology, object=URIRef(url))
+
+    def __str__(self):
+        return self.convert_to_rdf(self.docs).serialize(format='turtle').decode("utf-8")
+
+    def __unicode__(self):
+        return self.convert_to_rdf(self.docs).serialize(format='turtle').decode("utf-8")
+
+    def write_file(self, filename):
+        if sys.version_info < (3,):
+            data = unicode(self).encode('utf-8')
+        else:
+            data = str(self)
+
+        f = open(filename, "w")
+        f.write(data)
+        f.close()
+
+
+if __name__ == "__main__":
+    l1 = "./python-odml/doc/example_odMLs/ex_1.odml"
```

Comment:
Not sure what this main function is for. Is it for local testing during development? I wouldn't want it merged into master, but I guess we can leave it for the dev branch (?)